### PR TITLE
Abstract ApiController

### PR DESCRIPTION
**Pull request by poppabear8883**
This proposes that we add this abstract controller do reduce the code
duplication. Create new controllers on the fly by setting a few values
and moving on with life.

In addition, We could also extend the `make:controller` Artisan command to include something like .. `make:controller --api` and using a stub file that will generate these controllers.

Maybe the Artisan command should be included in this PR ?

### CODE REVIEW

File: app/Http/Requests/UpdateArtist.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'moniker' => 'required|max:255', // TODO: removing `required` from this causes tests to fail ?
+            'alt_moniker' => 'max:255',
+            'email' => 'email',
```
Comment (cbj4074) [inline app/Http/Requests/UpdateArtist.php]: Seems like this was added accidentally. Can we remove it?

Comment (poppabear8883) [reply app/Http/Requests/UpdateArtist.php]: I never added that, that was what you had in `StoreArtist` and you was using `StoreArtist` for the `update()` method, so i simply made a new `UpdateArtist` and put the same rules in there ... 

but either way if the email is passed with an update request it should in  fact be validated as valid email no ? I think all possible fields need a minimal validation even if we are just setting a `min:` and/or `max:` ...

we should ***not*** remove the `email` validation *just* to please a test ...

Comment (cbj4074) [reply app/Http/Requests/UpdateArtist.php]: My bad, you're absolutely correct here. I forgot that I had added email as a public profile field. It doesn't exist on the legacy site, and I forgot that I had added it intentionally.

File: app/Http/Requests/UpdateArtist.php [hunk 2]
```diff
@@ -24,7 +24,6 @@ public function authorize()
     public function rules()
     {
         return [
-            'moniker' => 'required|max:255', // TODO: removing `required` from this causes tests to fail ?
```
Comment (poppabear8883) [inline app/Http/Requests/UpdateArtist.php]: Shouldn't this remain here ? Just ***not*** `required` ?

File: tests/Feature/Controllers/ArtistControllerTest.php [hunk 3]
```diff
@@ -19,6 +31,13 @@ public function setUp()
         $this->country = resolve(CountryRepositoryInterface::class);
         $this->artist = resolve(ArtistRepositoryInterface::class);
         $this->profile = resolve(ProfileRepositoryInterface::class);
```
Comment (poppabear8883) [inline tests/Feature/Controllers/ArtistControllerTest.php]: Shouldn't all 3 of these properties be initialized above just like the below ones ?

Comment (cbj4074) [reply tests/Feature/Controllers/ArtistControllerTest.php]: Yes; but a bunch of tests need the same type of clean-up, so I'm going to handle them in a single commit.

File: app/Http/Requests/UpdateArtist.php (hunk 2 above)
Comment (poppabear8883) [reply app/Http/Requests/UpdateArtist.php]: Oh I read the commit body ... sorry for missing that

Comment (cbj4074) [reply app/Http/Requests/UpdateArtist.php]: That's a good point; even though Catalog Entities (Artists and Labels) aren't permitted to change their own monikers (they must submit a request for same, which a staff member must review prior to accepting the change), the validation rules should still apply. I'll fix this now.

**Comment by poppabear8883** [comment]
Still needs a few things before we close this out:

- [ ] Extend `make:controller` to include `--api`
- [ ] Create Api Controller stub file

**Comment by cbj4074** [comment]
Great work on this, @poppabear8883 !

**Pull request merged**